### PR TITLE
Fix NPE and build error on Windows

### PR DESCRIPTION
**Pull request by dwickern**

### CODE REVIEW

File: src/main/scala/io/swagger/scala/converter/SwaggerScalaModelConverter.scala [hunk 1]
```diff
@@ -55,21 +55,25 @@ class SwaggerScalaModelConverter extends ModelConverter {
...
           Option(resolveProperty(nextType, context, annotations, chain)) match {
             case Some(p) => Some(p)
-            case None if chain.hasNext() =>
+            case None if chain.hasNext =>
               Option(chain.next().resolveProperty(nextType, context, annotations, chain))
             case _ => None
           }
         }
         nextResolved match {
-          case Some(nextResolved) =>
```
Comment (fehguy) [inline src/main/scala/io/swagger/scala/converter/SwaggerScalaModelConverter.scala:64]: Yes, this is horrible :)  Thanks for changing

File: src/main/scala/io/swagger/scala/converter/SwaggerScalaModelConverter.scala [hunk 2]
```diff
@@ -55,21 +55,25 @@ class SwaggerScalaModelConverter extends ModelConverter {
...
+          case Some(property) =>
+            property.setRequired(false)
+            property
           case None => null
         }
       case t if chain.hasNext =>
-        val nextResolved = chain.next().resolveProperty(t, context, annotations, chain)
-        nextResolved.setRequired(true)
-        nextResolved
+        val nextResolved = Option(chain.next().resolveProperty(t, context, annotations, chain))
```
Comment (fehguy) [inline src/main/scala/io/swagger/scala/converter/SwaggerScalaModelConverter.scala:70]: Good find

**Review by fehguy — APPROVED** [review]

**Comment by dwickern** [comment]
I'm not sure why I'm working on this `develop` branch, want me to rebase on master?

**Comment by pjfanning** [comment]
@dwickern would it be possible to add a test case?

**Comment by dwickern** [comment]
OK, the test is pretty contrived but reproduces the problem. In practice I registered my custom model converter in META-INF. Can't do that in tests because it will affect all tests on the classpath

**Comment by dwickern** [comment]
Is there anything more I can do to get this merged?

**Comment by dwickern** [comment]
I published this to my namespace if anyone needs
```
excludeDependencies += "io.swagger" %% "swagger-scala-module"
libraryDependencies += "com.github.dwickern" %% "swagger-scala-module" % "1.0.5-PR39"
```

**Comment by pjfanning** [comment]
@webron @frantuma @fehguy can we get this merged?

**Comment by dwickern** [comment]
Still waiting to get this merged 😄

**Pull request merged**